Spigot-API-Patches/0291-add-RespawnFlags-to-PlayerRespawnEvent.patch

```diff
@@ -0,0 +1,73 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Thu, 22 Apr 2021 17:17:54 -0700
+Subject: [PATCH] add RespawnFlags to PlayerRespawnEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java b/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java
+index d2be2ad2e3665728e614a89dd62ef9237f1d3ce6..4e85daf5b4f8cc77dfcfd57306fd805e4a4905df 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java
+@@ -14,17 +14,30 @@ public class PlayerRespawnEvent extends PlayerEvent {
+     private Location respawnLocation;
+     private final boolean isBedSpawn;
+     private final boolean isAnchorSpawn;
++    private final java.util.Set<RespawnFlag> respawnFlags; // Paper
+ 
+     @Deprecated
+     public PlayerRespawnEvent(@NotNull final Player respawnPlayer, @NotNull final Location respawnLocation, final boolean isBedSpawn) {
+         this(respawnPlayer, respawnLocation, isBedSpawn, false);
+     }
+ 
++    @Deprecated // Paper
+     public PlayerRespawnEvent(@NotNull final Player respawnPlayer, @NotNull final Location respawnLocation, final boolean isBedSpawn, final boolean isAnchorSpawn) {
++        // Paper start
++        this(respawnPlayer, respawnLocation, isBedSpawn, isAnchorSpawn, com.google.common.collect.Sets.newHashSet());
++    }
++
++    public PlayerRespawnEvent(@NotNull final Player respawnPlayer, @NotNull final Location respawnLocation, final boolean isBedSpawn, final boolean isAnchorSpawn, @NotNull final java.util.Set<RespawnFlag> respawnFlags) {
++        // Paper end
+         super(respawnPlayer);
+         this.respawnLocation = respawnLocation;
+         this.isBedSpawn = isBedSpawn;
+         this.isAnchorSpawn = isAnchorSpawn;
++        // Paper start
++        this.respawnFlags = respawnFlags;
++        if (this.isBedSpawn) { this.respawnFlags.add(RespawnFlag.BED_SPAWN); }
++        if (this.isAnchorSpawn) { this.respawnFlags.add(RespawnFlag.ANCHOR_SPAWN); }
++        // Paper end
+     }
+ 
+     /**
+@@ -77,4 +90,31 @@ public class PlayerRespawnEvent extends PlayerEvent {
+     public static HandlerList getHandlerList() {
+         return handlers;
+     }
++
++    // Paper start
++    /**
++     * Gets whether the respawn event was caused by the end portal.
```

Comment:
The javadocs should probably be updated to reflect on the `RespawnFlag` concept.
As the collection seems mutable (tho modifications are not respected) this should also be stated in the documentation.

Reply:
I guess the collection doesn’t have to be mutable. Changing them doesn’t do anything, at the moment it’s only an informative set of flags.

Reply:
In my opinion it does not really matter whether or not the collection is immutable, just wanted to have the documentation state it 😂

Reply:
so ya, did that. Also changed it to an immutable set.